drivers/i2c/i2c_common.c

```diff
@@ -26,19 +26,27 @@ void z_i2c_transfer_signal_cb(const struct device *dev,
 #endif
 
 void i2c_dump_msgs(const char *name, const struct i2c_msg *msgs,
-		   uint8_t num_msgs, uint16_t addr)
+		   uint8_t num_msgs, uint16_t addr, bool dump_read)
```

Comment:
Can we separate the changes adding a new parameter from the reformatting of the logging?

Reply:
They are two separate commits.

Reply:
From 9739a14017686b910ec5de193ea17be99f6a57f6:

Commit Message:

> i2c: Improve formatting of i2c dump function
If an i2c message is for just one byte, instead of logging it with a
hexdump after logging the other message info, just added the one byte to
the same log message.

Since most i2c messages are one byte, this significantly reduces the
number of messages and lines needed to log i2c transactions, from three
line per message to just one.  It's also a lot easier to read.

Also add an option to log the data from reads too.  If the dump is done
after a transaction is processed, as opposed to before, then the read
data is valid and can be very useful.

9739a14017686b9 is adding the parameter to log the data from reads in addition to reformatting the output.

Reply:
Oh, I thought you meant the new configuration option should be a different commit, which it is.

I'd have to write another version of the logging function since the logic for both changes touch every line.  Seems like a lot of trouble, to write another function just to delete it in the next patch.  And of course this version will not be tested at all.

Reply:
I suppose it's a simple enough commit that this is fine. I just very much prefer to see diffs that are specific to each change.